File: cases/taint/javascript/model-opaque-propagator-negative/Opaque.js
function dfb_source() { // DFB-SOURCE: model-opaque-propagator-input
  return "tainted";
}

function dfb_sink(value) {} // DFB-SINK: model-opaque-propagator-sink

const _impl = {
  identity: function identity(value) {
    return value;
  }
};

const Opaque = {
  carry: function carry(value) {
    const target = "identity";
    return Reflect.get(_impl, target).apply(null, [value]);
  },
  block: function block(value) {
    const target = "identity";
    return Reflect.get(_impl, target).apply(null, [value]);
  }
};

function run() {
  dfb_sink(Opaque.block(dfb_source()));
}

File: cases/taint/python/model-opaque-propagator-positive/opaque.py
class _Impl:
    @staticmethod
    def identity(value):
        return value


_impl = _Impl()


def carry(value):
    name = "identity"
    return getattr(_impl, name)(value)


def block(value):
    name = "identity"
    return getattr(_impl, name)(value)


def dfb_source():  # DFB-SOURCE: model-opaque-propagator-input
    return "tainted"


def dfb_sink(value):  # DFB-SINK: model-opaque-propagator-sink
    pass


def run():
    dfb_sink(carry(dfb_source()))
